api/app/Http/Controllers/Api/AuthSocialiteController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\User;
+use Tymon\JWTAuth\Facades\JWTAuth;
+use App\Http\Controllers\Controller;
+use Laravel\Socialite\Facades\Socialite;
+
+class AuthSocialiteController extends Controller
+{
+    private $driver = 'facebook';
+
+    public function redirectToProvider()
+    {
+        return Socialite::driver($this->driver)->stateless()->redirect();
+    }
+
+    public function handleProviderCallback()
+    {
+        $facebookUser = Socialite::driver($this->driver)->stateless()->user();
+
+        $user = User::where('email', $facebookUser->email)
+            ->where('provider', $this->driver)
+            ->where('provider_id', $facebookUser->id)
+            ->first();
+
```

Comment:
What if the user is already in the database with his/her facebook email? Create it again?